### PR TITLE
Update config docs, add CI/CD and Versions guide, add CI check for broken links

### PR DESCRIPTION
**Pull request by jonaro00**

### CODE REVIEW

File: .github/workflows/ci.yml
[the remark this answers is not in the document]
Comment (jonaro00) [reply .github/workflows/ci.yml:1]: Different repo. And this is free 😄

[the remark this answers is not in the document]
Comment (orhun) [reply .github/workflows/ci.yml:1]: Ah, damn. Got it xD

File: configuration/logs.mdx [hunk 1]
```diff
@@ -33,19 +36,22 @@ in your project is add it to your `Cargo.toml`.
...
+We have a few commands that are useful for viewing your application's logs.
 
-To view the logs for your current deployment, in your project directory run:
+To view the logs for your current `running` deployment, if there is one:
 
 ```bash
 cargo shuttle logs
 ```
 
+*Caveat: The above command finds the oldest deployment in the `running` state (there should only be one...).*
```
Comment (oddgrd) [inline configuration/logs.mdx:53]: I may be overthinking this, but I feel this part makes it seem like we don't take this bug seriously, which could be frustrating for users. If we keep this caveat here I think we should rather link to an issue tracking it, or a faq entry, where they can find how to work around it as well.
```suggestion
*Caveat: The above command finds the oldest deployment in the `running` state*
```

File: configuration/shuttle-versions.mdx
Comment (oddgrd) [inline configuration/shuttle-versions.mdx:1]: Nice!

**Review by oddgrd — COMMENTED** [review]
Looks great to me, just one small comment.

**Review by oddgrd — APPROVED** [review]

**Pull request merged**